app/Http/Controllers/Order/RenewController.php

```diff
@@ -304,7 +305,22 @@ public function renewForm($id, $agents = null)
 
     public function renewByClient($id, Request $request)
     {
-        $userId = Subscription::find($id)->user_id;
+        $subscription = Subscription::find($id);
```

Comment:
$subscription = Subscription::find($id);

$latestInvoiceItem = InvoiceItem::where('plan_id', $request->plan)
    ->whereHas('invoice', function ($query) {
        $query->where('is_renewed', 1)->where('status', 'pending');
    })
    ->latest('created_at')
    ->first();

if ($latestInvoiceItem) {
    return redirect()->route('my-invoice', ['id' => $latestInvoiceItem->invoice_id])
        ->with('warning', 'You have an existing unpaid invoice for this product. Please proceed with the payment or delete the invoice and try again');
}

Move this also to a different private method